src/libflake-c/nix_api_flake.cc

```diff
@@ -206,4 +206,20 @@ nix_value * nix_locked_flake_get_output_attrs(
     NIXC_CATCH_ERRS_NULL
 }
 
+nix_err nix_locked_flake_read_path(
+    nix_c_context * context,
+    nix_locked_flake * lockedFlake,
+    const char * path,
+    nix_get_string_callback callback,
+    void * user_data)
+{
+    nix_clear_err(context);
+    try {
+        auto source_path = lockedFlake->lockedFlake->flake.path.parent() / path;
```

Comment:
```suggestion
        auto source_path = lockedFlake->lockedFlake->flake.path.parent() / CanonPath(path);
```
Otherwise I think it may select the wrong overload for the `/` operator, namely:
```
    /**
     * Append a single component `c` to this path. `c` must not
     * contain a slash. A slash is implicitly added between this path
     * and `c`.
     */
    SourcePath operator/(std::string_view c) const;
```
so appending a path with multiple components (`foo/bar.nix`) wouldn't work.